c7n/ctx.py

```diff
@@ -26,43 +35,87 @@ def __init__(self, session_factory, policy, options):
         self.policy = policy
         self.options = options
         self.session_factory = session_factory
-        self.cloudwatch_logs = None
+
+        # Runtime initialized during policy execution
+        # We treat policies as a fly weight pre-execution.
         self.start_time = None
+        self.execution_id = None
+        self.output = None
+        self.api_stats = None
+        self.sys_stats = None
+
+        # A few tests patch on metrics flush
+        self.metrics = metrics_outputs.select(self.options.metrics_enabled, self)
 
-        metrics_enabled = getattr(options, 'metrics_enabled', None)
-        factory = MetricsOutput.select(metrics_enabled)
-        self.metrics = factory(self)
+        # Tracer is wired into core filtering code / which is getting
+        # invoked sans execution context entry in tests
+        self.tracer = tracer_outputs.select(self.options.tracer, self)
```

Comment:
update those tests to use initialize, but its unclear if we wouldn't want the behavior their doing to work ootb, namely use a policy to filter a set of resources sans exec.